src/library_html5.js

```diff
@@ -2598,6 +2598,157 @@ var LibraryJSEvents = {
 
   emscripten_html5_remove_all_event_listeners: function() {
     JSEvents.removeAllEventListeners();
+  },
+
+  emscripten_request_animation_frame: function(cb, userData) {
+    return requestAnimationFrame(function(timeStamp) {
+      Module['dynCall_idi'](cb, timeStamp, userData);
```

Comment:
this should check if `ABORT` (whether the program has halted). `Browser.safeRequestAnimationFrame` does that. So perhaps we should move that to a toplevel function outside of Browser (for code size) and use it from here and inside Browser?

Reply:
These functions are used in scenarios where `library_browser.js` is not linked in to the program at all, and the variable `ABORT` does not exist in the generated program. The intent of these is to give a direct JS passthrough to the functions in a way that does not build up abstractions beyond the minimum. Checking for `ABORT` would just increase redundant code size, so I would not prefer to add that handling here. In practice when/if something throws from within the rAF callback, execution will unwind past this handler, so the next requestAnimationFrame will not be fired registered anyways.

Reply:
ABORT should always be present, no? Or is this with custom JS prelude code?

How about adding an option (sort of related to EXIT_RUNTIME, but not identical) that indicates whether ABORT should be checked? Then we can skip it here and elsewhere.

Reply:
To add some detail, it can be extremely confusing for an application to hit an exception and be supposed to halt, but existing JS callbacks keep things running. In JS, promises can sort of automatically avoid it. For emscripten C code, we use that ABORT global.

I do see your point about code size, so I think `CAN_ABORT=0` for your use case would be a reasonable way to emit minimal code. But by default I think it would be better for users to have premature halting handled properly, and it would be consistent with our other logic elsewhere too.

Reply:
ABORT does not exist in the custom environment that we are using, but it gets optimized away. In development builds I can appreciate the usefulness. The idea with these callbacks is to not start building up POSIX-like abstractions, but just give the user the minimum import of the functionality that was requested.

I don't think it will be surprising to get an exception and keep on running event callbacks, because the semantics without an ABORT are identical to what happens in JavaScript already? That is, if one registers a setInterval loop, then the setInterval will keep firing even if each event handler invocation throws. Likewise, if one does requestAnimationFrame()s, setTimeout()s, setImmediate()s or similar and the function that registered them throws, then those pending event handlers will still fire, and that is what JS developers already know will happen. Not many web pages have a concept of the page being alive or dead, but the page is treated to be always there - i.e. this is departing away from the POSIX like model.

Anyhow, added a `-s CAN_ABORT=0/1` option, defaulting to enabled, if this is important, I can easily turn it off in my own builds.

Reply:
Yeah, I see what you mean better now. I guess I'm convinced enough to go either way on this.

If we go your way and don't use ABORT here, then I think we should document those functions that way, that is, mention something like `these are low-level functions that directly call the relevant Web APIs. They don't handle integration with the emscripten runtime (such as checking if the program has halted and cancelling a callback if so).`

Or if we do go with `ABORT` here, then I had a suggestion for rephrasing the `settings.js` comment, and also, the new `CAN_ABORT` option should be used in all other relevant places too (i.e., no need to emit any code with `ABORT` if `CAN_ABORT=0`).

I slightly lean towards your side at this point, so the former option. What do you think?

Reply:
Another point in favor of your approach is that we should make `ASSERTIONS=1` mode have checks for calling compiled code after an ABORT. That would make debugging such things much easier. I can look into that.

Reply:
Ok, removed the `CAN_ABORT` bit for now. Indeed there will be many other places where such a CAN_ABORT option would touch, so that could be a set of work on its own.

Reply:
Added the above note to the documentation page.